qiskit_nature/transformers/second_quantization/electronic/seniority_zero_transformer.py

```diff
@@ -0,0 +1,246 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Seniority-Zero Restriction interface."""
+
+from copy import deepcopy
+import logging
+
+from functools import partial
+from typing import cast, Tuple
+import numpy as np
+
+from qiskit_nature import QiskitNatureError, ListOrDictType, settings
+from qiskit_nature.operators.second_quantization import VibrationalOp
+from qiskit_nature.properties import GroupedProperty, Property
+from qiskit_nature.properties.second_quantization import (
+    GroupedSecondQuantizedProperty,
+    SecondQuantizedProperty,
+)
+from qiskit_nature.properties.second_quantization.driver_metadata import DriverMetadata
+from qiskit_nature.properties.second_quantization.electronic import (
+    ParticleNumber,
+    ElectronicEnergy,
+)
+from qiskit_nature.properties.second_quantization.electronic.bases import (
+    ElectronicBasisTransform,
+    ElectronicBasis,
+)
+from qiskit_nature.properties.second_quantization.electronic.types import GroupedElectronicProperty
+from qiskit_nature.transformers.second_quantization import BaseTransformer
+
+logger = logging.getLogger(__name__)
+
+
+class SeniorityZeroTransformer(BaseTransformer):
+    """The Seniority-Zero restriction."""
+
+    def __init__(self):
+        self._num_modes: int = None
+
+    def transform(
```

Comment:
Are there any ordering/usage constraints when using this with other Transformers? As it stands the problem takes a list of transformers and they are simply applied one after the other in list order.

Reply:
Yes, this transformer would need to be applied after any active space or freeze core transformer. Once we get a particle hole transformer, I believe that will be incompatible to be matched with this one. @pbark can correct me on this if needed

Reply:
Should we be thinking about say a priority attribute such that the list could be processed in priority order too? That way the user would not have to be so concerned but can just say I want these transformations and the list order plus priority order deals with things. They would still need of course to know that when using SZ transform that they will need to alter the mapping according. But dealing with list order would be more intrinsic. At present, if they put it up front in the list, what would happen - do we check anywhere or do things just fail somehow later?

Reply:
Yes this sounds like a good idea. But I think for now we can merge this as is and implement a priority ordering in a separate PR. If you agree I can open an issue to track this :+1: